kernel/boot/Handoff.cpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file Handoff.cpp
+ * @author Keeton Feave (keetonfeavel@cedarville.edu)
+ * @brief 
+ * @version 0.1
+ * @date 2021-06-02
+ * 
+ * @copyright Copyright the Panix Contributors (c) 2021
+ * 
+ */
+#include <boot/Handoff.hpp>
+#include <lib/stdio.hpp>
+#include <lib/string.hpp>
+#include <mem/paging.hpp>
+#include <dev/serial/rs232.hpp>
+
+#include <multiboot/multiboot2.h>
+#include <stivale/stivale2.h>
+
+namespace Boot {
+
+/*
+ *  ___                   _          __  __
+ * | __| _ __ _ _ __  ___| |__ _  _ / _|/ _|___ _ _
+ * | _| '_/ _` | '  \/ -_) '_ \ || |  _|  _/ -_) '_|
+ * |_||_| \__,_|_|_|_\___|_.__/\_,_|_| |_| \___|_|
+ *
+ */
+
+FramebufferInfo::FramebufferInfo()
+    : _addr(NULL)
+    , _width(0)
+    , _height(0)
+    , _depth(0)
+    , _pitch(0)
+    , _redMaskSize(0)
+    , _redMaskShift(0)
+    , _greenMaskSize(0)
+    , _greenMaskShift(0)
+    , _blueMaskSize(0)
+    , _blueMaskShift(0)
+    , _memoryModel(Undefined_FBMM)
+{
+    // Default constructor.
+}
+
+FramebufferInfo::FramebufferInfo(uint16_t width, uint16_t height, uint16_t depth, uint16_t pitch, void* addr)
+    : _addr(addr)
+    , _width(width)
+    , _height(height)
+    , _depth(depth)
+    , _pitch(pitch)
+{
+    // TODO: Can we call the default constructor or will that overwrite
+    // the values passed in the initialization list?
+    _redMaskSize = 0;
+    _redMaskShift = 0;
+    _greenMaskSize = 0;
+    _greenMaskShift = 0;
+    _blueMaskSize = 0;
+    _blueMaskShift = 0;
+    _memoryModel = Undefined_FBMM;
+}
+
+FramebufferInfo::~FramebufferInfo()
+{
+    // Nothing to deconstruct
+}
+
+uint16_t FramebufferInfo::getWidth() { return _width; }
+uint16_t FramebufferInfo::getHeight() { return _height; }
+uint16_t FramebufferInfo::getDepth() { return _depth; }
+
+/*
+ *  _  _              _      __  __
+ * | || |__ _ _ _  __| |___ / _|/ _|
+ * | __ / _` | ' \/ _` / _ \  _|  _|
+ * |_||_\__,_|_||_\__,_\___/_| |_|
+ *
+ */
+
+Handoff::Handoff(void* handoff, uint32_t magic)
+    : _handle(handoff)
+    , _magic(magic)
+{
+    // Parse the handle based on the magic
+    if (magic == 0x36d76289) {
+        _bootType = Multiboot2;
+        parseMultiboot2(handoff);
+    } else if (magic == *(uint32_t*)"stv2") {
+        _bootType = Stivale2;
+        parseStivale2(handoff);
+    }
+}
+
+Handoff::~Handoff()
+{
+    // Nothing to deconstruct
+}
+
+void Handoff::parseStivale2(void* handoff)
```

Comment:
Can we move this to a static builder method where it constructs and returns a `Handoff` object?

Reply:
The idea is that the constructor takes the `void* handoff` pointer and then will call either `parseStivale2` or `parseMultiboot2` and construct itself that way, so basically the same end result.

Reply:
Oh. Calling member functions from the constructor. Nice.

Reply:
Updated in 651d0e3. Let me know if you have any other thoughts on the matter.